daml-script/runner/src/main/scala/com/digitalasset/daml/lf/engine/script/v2/ScriptF.scala

```diff
@@ -85,11 +93,61 @@ object ScriptF {
       valueTranslator.translateValue(ty, value).left.map(_.toString)
 
   }
-  final case class Submit(data: SubmitData) extends Cmd {
-    override def stackTrace = data.stackTrace
 
-    override def description = "submit"
+  final case class Throw(exc: SAny) extends Cmd {
+    override def execute(
+        env: Env
+    )(implicit ec: ExecutionContext, mat: Materializer, esf: ExecutionSequencerFactory) =
+      Future.failed(
+        script.Runner.InterpretationError(
+          SError
+            .SErrorDamlException(IE.UnhandledException(exc.ty, exc.value.toUnnormalizedValue))
+        )
+      )
+  }
+
+  final case class Catch(act: SValue) extends Cmd {
+
+    val left = SEBuiltin(
+      SBVariantCon(StablePackage.DA.Types.Either, Name.assertFromString("Left"), 0)
+    )
+    val right = SEBuiltin(
+      SBVariantCon(StablePackage.DA.Types.Either, Name.assertFromString("Right"), 1)
+    )
+
+    override def executeWithRunner(env: Env, runner: v2.Runner)(implicit
+        ec: ExecutionContext,
+        mat: Materializer,
+        esf: ExecutionSequencerFactory,
+    ): Future[SExpr] =
+      runner.runExpr(SEAppAtomic(SEValue(act), Array(SEValue(SUnit)))).transformWith {
+        case Success(v) =>
+          Future.successful(SEAppAtomic(right, Array(SEValue(v))))
+        case Failure(
+              script.Runner.InterpretationError(
+                SError.SErrorDamlException(IE.UnhandledException(typ, value))
+              )
+            ) =>
+          Future.successful(
+            SELet1(
+              SEImportValue(typ, value),
+              SELet1(
+                SEAppAtomic(SEBuiltin(SBToAny(typ)), Array(SELocS(1))),
+                SEAppAtomic(left, Array(SELocS(1))),
+              ),
+            )
+          )
+        case Failure(e) => Future.failed(e)
+      }
+
+    override def execute(env: Env)(implicit
+        ec: ExecutionContext,
+        mat: Materializer,
+        esf: ExecutionSequencerFactory,
+    ): Future[SExpr] = Future.failed(new NotImplementedError)
```

Comment:
Is this the correct error? `executeWithRunner` by default defers to `execute`, but we don't here.